File: src/main/java/uk/co/lukestevens/logging/LoggingProvider.java
package uk.co.lukestevens.logging;

/**
 * An interface for defining a provider of
 * Loggers.
 * 
 * @author Luke Stevens
 */
public interface LoggingProvider {
	
	/**
	 * @param name The name of the logger
	 * @return A new logger with the given name
	 */
	Logger getLogger(String name);
	
	/**
	 * @param className The class this logger is for
	 * @return A new logger with the name of the class
	 */
	default Logger getLogger(Class<?> className) {
		return this.getLogger(className.getName());
	}

}

File: src/main/kotlin/uk/co/lukestevens/logging/LoggingProvider.kt
package uk.co.lukestevens.logging

/**
 * An interface for defining a provider of
 * Loggers.
 *
 * @author Luke Stevens
 */
interface LoggingProvider {
	/**
	 * @param name The name of the logger
	 * @return A new logger with the given name
	 */
	fun getLogger(name: String): Logger

	/**
	 * @param className The class this logger is for
	 * @return A new logger with the name of the class
	 */
	fun getLogger(className: Class<*>): Logger {
		return this.getLogger(className.getName())
	}
}